### ISSUE TITLE
False positive explicit-length-check for 0 === x.length

### ISSUE THREAD

**Issue by robopzet** [sindresorhus/eslint-plugin-unicorn#1749]
The  `explicit-length-check` produces a warning when checking for `0` with code in yoda style:

```
if (0 === doms.length) {
```

When the values are switched the warning does not appear (but that produces a `yoda` error).

```
if (doms.length === 0) {
```

Rule: explicit-length-check

An other example:

```js
if (0 === qryrslt.dta[i].length) return false
```

**Comment by fisker** [comment]
This is expected, we don't allow "yoda", see 4th example here https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/explicit-length-check.md#fail

**Comment by robopzet** [comment]
OK, I didn't see that example. Also did not expect it. I assumed the rule only flags the comparison with `.length`, not force other coding rules.

**Comment by fisker** [comment]
If you really want it to work with `yoda`, you can propose a rule change to disscuss. Closing now.
